### PR TITLE
Add leave share menu item

### PR DESCRIPTION
**Pull request by haidian6672**

### CODE REVIEW

**Comment by haidian6672** [comment]
default

**Comment by haidian6672** [comment]
default

File: src/ui/repo-tree-view.cpp [hunk 1]
```diff
@@ -428,6 +432,12 @@ void RepoTreeView::createActions()
 
     connect(open_in_filebrowser_action_, SIGNAL(triggered()), this, SLOT(openInFileBrowser()));
 
+    unshare_action_ = new QAction(tr("&Leave share"), this);
+    unshare_action_->setIcon(QIcon(":/images/leave-share.png"));
+    unshare_action_->setStatusTip(tr("unshare this library"));
```
Comment (lins05) [inline src/ui/repo-tree-view.cpp]: "unshare this library"  is improper here. Just use "leave share".

File: src/ui/repo-tree-view.cpp [hunk 2]
```diff
@@ -571,6 +581,32 @@ void RepoTreeView::shareRepoToGroup()
...
 
+void RepoTreeView::unshareRepo()
+{
+    const Account account = seafApplet->accountManager()->currentAccount();
+    const QString repo_id = selected_repo_.id;
+    const QString from_user = selected_repo_.owner;
+    UnshareRepoRequest* request =
+        new UnshareRepoRequest(account, repo_id, from_user);
+
+    connect(request, SIGNAL(success()),
```
Comment (lins05) [inline src/ui/repo-tree-view.cpp]: Should we show an error message to the user when the operation fails?

**Comment by shoeper** [comment]
Is there a confirmation dialog? I suggest to add one if not because clicking on a menu item can happend by accident easily. ` No` should be the default of the dialog.

**Comment by lins05** [comment]
@shoeper That makes sense. 

@haidian6672  Let's ask the user to confirm before sending the request.

File: src/ui/repo-tree-view.cpp [hunk 3]
```diff
@@ -583,6 +583,12 @@ void RepoTreeView::shareRepoToGroup()
 
 void RepoTreeView::unshareRepo()
 {
+    if (!seafApplet->yesOrNoBox(
+            tr("Are you sure to leave share \"%1\"?").arg(
```
Comment (shoeper) [inline src/ui/repo-tree-view.cpp]: Are you sure you want to leave the share

File: src/ui/repo-tree-view.cpp [hunk 4]
```diff
@@ -607,6 +615,11 @@ void RepoTreeView::onUnshareSuccess()
     }
 }
 
+void RepoTreeView::onUnshareFailed(const ApiError&error)
+{
+    seafApplet->warningBox(tr("Leave share failed"), this);
```
Comment (shoeper) [inline src/ui/repo-tree-view.cpp]: Leaving share failed

**Pull request merged**